build-packages.sh

```diff
@@ -3,159 +3,65 @@
 usage()
 {
     echo "Builds the NuGet packages from the binaries that were built in the Build product binaries step."
-    echo "Usage: build-packages [arch] [configuration]"
+    echo "Usage: build-packages -BuildArch -BuildType"
     echo "arch can be x64, x86, arm, arm64 (default is x64)"
     echo "configuration can be release, checked, debug (default is debug)"
     echo
     exit 1
 }
 
 __ProjectRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-build_packages_log=$__ProjectRoot/build-packages.log
-binclashlog=$__ProjectRoot/binclash.log
-binclashloggerdll=$__ProjectRoot/Tools/Microsoft.DotNet.Build.Tasks.dll
-RuntimeOS=ubuntu.$VERSION_ID
 
-__MSBuildPath=$__ProjectRoot/Tools/MSBuild.exe
-
-# Parse arguments
-__BuildArch=x64
-__BuildType=Debug
-
-allargs="$@"
-
-echo -e "Running build-packages.sh $allargs" > $build_packages_log
-
-if [ "$allargs" == "-h" ] || [ "$allargs" == "--help" ]; then
-    usage
-fi
+unprocessedBuildArgs=
 
 while :; do
     if [ $# -le 0 ]; then
         break
     fi
 
-    lowerI="$(echo $1 | awk '{print tolower($0)}')"
-    case $lowerI in
+	opt="$1"
+    case $opt in
         -\?|-h|--help)
-            usage
-            exit 1
-            ;;
-
-        x86)
-            __BuildArch=x86
-            ;;
-
-        x64)
-            __BuildArch=x64
-            ;;
-
-        arm)
-            __BuildArch=arm
-            ;;
-
-        arm64)
-            __BuildArch=arm64
-            ;;
-        debug)
-            __BuildType=Debug
-            ;;
-        release)
-            __BuildType=Release
-            ;;
-        checked)
-            __BuildType=Checked
+        usage
```

Comment:
Is the idea of the loop to get rid of the parameter `-?` I don't see what else are you taking out of `$*`. 
If this is the case you could ask 
```
if [ $1 == "-?" ]; then
    usage
fi
```
And then just pass `$*` to the project calls